lib/bpk-component-calendar/src/common-types.js

```diff
@@ -30,13 +30,120 @@ import { type FooterView } from './footerView';
 type ViewProps = ElementProps<typeof View>;
 type ViewStyleProp = $PropertyType<ViewProps, 'style'>;
 
-export const SELECTION_TYPES = {
-  single: 'single',
-  range: 'range',
-  multiple: 'multiple',
+export type SelectionTypeSingle = {|
+  type: 'single',
+
+  // A hint that is read out to screen reader users
+  // when focus is on an unselected date. Should explain how
+  // to select the date i.e. by double tapping.
+  selectHint: string,
+|};
+
+export type SelectionTypeRange = {|
+  type: 'range',
+
+  // A hint that is read out to screen reader users
+  // when no dates have been selected. Should explain how
+  // to select the start date i.e. by double tapping.
+  startDateSelectHint: string,
+
+  // A hint that is read out to screen reader users
+  // when only the start date has been selected. Should explain how
+  // to select the end date i.e. by double tapping.
+  endDateSelectHint: string,
+
+  // A string that is read out to screen reader users
+  // when the selected start date has focus.
+  startDateSelectedState: string,
+
+  // A string that is read out to screen reader users
+  // when the selected end date has focus.
+  endDateSelectedState: string,
+
+  // A string that is read out to screen reader users
+  // when the selected end date is the same as the start date and
+  // this date has focus.
+  endAndStartDateSelectedState: string,
+
+  // A string that is read out to screen reader users
+  // when a date between the start and end date has focus. e.g.
+  // when the start date is 01/03/2021 and the end date is 05/03/2021
+  // and the date with focus is 03/03/2021.
+  dateBetweenStartAndEndSelectedState: string,
+
+  // A prompt that is read out to screen reader users
+  // when a start date is selected. Should explain to the user
+  // that they should select an end date for the range.
+  makeNextSelectionPrompt: string,
+|};
+export type SelectionTypeMultiple = {|
+  type: 'multiple',
+
+  // A hint that is read out to screen reader users
+  // when focus is on an unselected date. Should explain how
+  // to select the date i.e. by double tapping.
+  selectHint: string,
+
+  // A hint that is read out to screen reader users
+  // when focus is on a selected date. Should explain how
+  // to unselect the date i.e. by double tapping.
+  deselectHint: string,
+|};
+
+export type SelectionType =
+  | SelectionTypeSingle
+  | SelectionTypeRange
+  | SelectionTypeMultiple;
+
+export const assertString = (
```

Comment:
I don't know if this is too heavy handed or not. WDYT?

The intent here is to catch any developers who ignore Flow. In a "proper" statically typed language this wouldn't be needed but since Flow errors can simply be ignored it might help

Reply:
I don't think it's too heavy handed. We know that consumers often ignore Flow types, and from an accessibility point of view that is unacceptable here, so we should enforce it. In other words, enforcing it is the most traveller-first thing to do.